ios_ui/MovieRec/RecommendationsTableViewController.swift

```diff
@@ -66,23 +70,62 @@ class RecommendationsTableViewController: UITableViewController {
         makeAPICall()
     }
     
-    private func refreshMovies(data: Data) -> Void {
-        let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
-        self.recommendations.removeAll()
-        if let responseJSON = responseJSON as? [String: [String]] {
-            for recommendation in responseJSON["recommendations"]! {
-                self.recommendations += [Recommendation(title: recommendation)]
-            }
+    private func refreshMovies(data: JSON) -> Void {
+        for recommendation in data.arrayValue {
+            self.recommendations += [Recommendation(title: recommendation.stringValue)]
+        }
 
-            DispatchQueue.main.async {
-                self.tableView.reloadData()
-            }
+        DispatchQueue.main.async {
```

Comment:
You probably want [unowned self] here so you dont run into retain cycles on this view controller